src/lightning/pytorch/trainer/connectors/data_connector.py

```diff
@@ -420,28 +420,17 @@ def _check_dataloader_iterable(
         )
 
 
-def _worker_check(trainer: "pl.Trainer", using_spawn: bool, dataloader: object, name: str) -> None:
+def _worker_check(trainer: "pl.Trainer", dataloader: object, name: str) -> None:
     if not isinstance(dataloader, DataLoader):
         return
 
     upper_bound = suggested_max_num_workers(trainer.num_devices)
+    using_spawn = mp.get_start_method() == "spawn"
```

Comment:
I'm slightly worried that `mp.get_start_method` wont return the proper start method for unconventional strategies (xla, habana, ...?). It might be safer to keep accessing the strategy for this

Reply:
Hmm that's ultimately how the DataLoader decides how to launch its workers. The dataloader also has a multiprocessing_context argument, we can take that into account: https://github.com/Lightning-AI/lightning/pull/18649/commits/400bd88f8d7f35c33a72275e8eb38ea678c4cf48

Note that our worker check here runs when processes have already been launched. Furthermore, it happens after the dataloader was processed by the strategy and for the XLA strategy, it is a [MpDeviceLoader](https://github.com/pytorch/xla/blob/1b68776e24ed90a5b48edbfca1e85b06128ab625/torch_xla/distributed/parallel_loader.py#L186)  which isn't a subclass of DataLoader and so we wouldn't emit a warning anyway.